### PR TITLE
Update user and create course category

### PR DESCRIPTION
**Pull request by LeoSYW**
1. Update user functions with Sequlize
2. Create Course Category functions and already tested by get and add.

### CODE REVIEW

File: appConfig.template.js
Comment (ZaneMa528) [inline appConfig.template.js:1]: Maybe consider keeping the appConfig.template.js. Therefore teammates can reclone this repo anytime.

File: router/coursecategoryrouter.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ *        description: Bad Request
+ *      404:
+ *        description: Not Found
+ *      409:
+ *        description: Conflict
+ *      500:
+ *        description: Server Error
+ */
+router.post(
+    "/addCourseCategory",
```
Comment (ZaneMa528) [inline router/coursecategoryrouter.js:53]: you already have the path /api/courseCategory, maybe consider remove addCourseCategory and let the HTTP method POST itself convey the action.

File: appConfig.template.js
Comment (LeoSYW) [reply appConfig.template.js:1]: I've reverted the rename change. Please have a check, thanks

File: router/coursecategoryrouter.js (hunk 1 above)
Comment (ZaneMa528) [reply router/coursecategoryrouter.js:53]: I see in the framework there are URL like adduser, and getuser.  maybe that's fine. We could discuss later

**Pull request closed without merging**